gradoop-data-integration/src/main/java/org/gradoop/dataintegration/transformation/EdgeToVertex.java

```diff
@@ -0,0 +1,206 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.dataintegration.transformation;
+
+import org.apache.flink.api.common.functions.FlatMapFunction;
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.util.Collector;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.impl.pojo.Edge;
+import org.gradoop.common.model.impl.pojo.EdgeFactory;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.common.model.impl.pojo.VertexFactory;
+import org.gradoop.flink.model.api.operators.UnaryGraphToGraphOperator;
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.functions.tuple.Value0Of3;
+
+import java.util.Objects;
+
+/**
+ * For edges of a specific label this graph transformation creates a new vertex containing the
+ * properties of the edge and two new edges respecting the direction of the original edge.
+ * The newly created edges and vertex labels are user-defined.
+ * <p>
+ * The original edges are still part of the resulting graph.
+ * Use a @{@link org.apache.flink.api.common.functions.FilterFunction} on the original label to
+ * remove them.
+ */
+public class EdgeToVertex implements UnaryGraphToGraphOperator {
+
+  /**
+   * The label of the edges use for the transformation.
+   */
+  private final String edgeLabel;
+
+  /**
+   * The label of the newly created vertex.
+   */
+  private final String newVertexLabel;
+
+  /**
+   * The label of the newly created edge which points to the newly created vertex.
+   */
+  private final String edgeLabelSourceToNew;
+
+  /**
+   * The label of the newly created edge which starts at the newly created vertex.
+   */
+  private final String edgeLabelNewToTarget;
+
+  /**
+   * The constructor for the structural transformation.
+   *
+   * @param edgeLabel            The label of the edges use for the transformation.
+   *                             (No edges will be transformed if this parameter is {@code null}).
+   * @param newVertexLabel       The label of the newly created vertex.
+   * @param edgeLabelSourceToNew The label of the newly created edge which points to the newly
+   *                             created vertex.
+   * @param edgeLabelNewToTarget The label of the newly created edge which starts at the newly
+   *                             created vertex.
+   *
+   * @throws NullPointerException when at least on of the latter 3 arguments is {@code null}.
+   */
+  public EdgeToVertex(String edgeLabel, String newVertexLabel, String edgeLabelSourceToNew,
+    String edgeLabelNewToTarget) {
+    Objects.requireNonNull(newVertexLabel);
```

Comment:
`requireNonNull` returns the object itself. You can use this in the assignment below.